formats/cbor/jvmTest/src/kotlinx/serialization/cbor/CborCompatibilityTest.kt

```diff
@@ -76,4 +79,42 @@ class CborCompatibilityTest {
         val serial = SomeComplexClass.serializer(ListSerializer(Int.serializer()))
         compare(obj, serial)
     }
+
+    @Test
+    fun testFloat() {
+        compare(FloatData(Float.NaN), FloatData.serializer())
+        compare(FloatData(Float.POSITIVE_INFINITY), FloatData.serializer())
+        compare(FloatData(Float.NEGATIVE_INFINITY), FloatData.serializer())
+        compare(FloatData(Float.MAX_VALUE), FloatData.serializer())
+        compare(FloatData(Float.MIN_VALUE), FloatData.serializer())
+        compare(FloatData(0.0f), FloatData.serializer())
+        compare(FloatData(-0.0f), FloatData.serializer())
+        compare(FloatData(-1.0f), FloatData.serializer())
+        compare(FloatData(1.0f), FloatData.serializer())
+        compare(FloatData(123.56f), FloatData.serializer())
+        compare(FloatData(123.0f), FloatData.serializer())
+        //minimal denormalized value in half-precision
+        compare(FloatData(5.9604645E-8f), FloatData.serializer())
```

Comment:
Am I right that the library we are testing our implementation with always encodes numbers as half-precision if this is possible to do so?

Reply:
Library tries to encode lossless by smallest format.
So double value can be encoded as double (as is), float (single precision) or half precision. Float can be sent as is or as half precision.